src/v/cloud_storage/remote_partition.cc

```diff
@@ -684,7 +687,7 @@ ss::future<storage::translating_reader> remote_partition::make_reader(
       "remote partition make_reader invoked, config: {}, num segments {}",
       config,
       _segments.size());
-    if (_segments.size() < _manifest.size()) {
+    if (static_cast<size_t>(_segments.size()) < _manifest.size()) {
```

Comment:
nit: Why did this change?  (& doesn't btree_map's size() already return a size_t?)

Reply:
it's a mystery to me, I noticed the warning and fixed it, seemed too minor for the separate commit

Reply:
i think static_cast should be removed. i mean, the mystery alone is worth investigating since it might point to something deeper.

```
[nwatkins@fedora clang]$ git diff
diff --git a/src/v/cloud_storage/remote_partition.cc b/src/v/cloud_storage/remote_partition.cc
index 7eea368e1..05d7e0ca2 100644
--- a/src/v/cloud_storage/remote_partition.cc
+++ b/src/v/cloud_storage/remote_partition.cc
@@ -687,7 +687,9 @@ ss::future<storage::translating_reader> remote_partition::make_reader(
       "remote partition make_reader invoked, config: {}, num segments {}",
       config,
       _segments.size());
-    if (static_cast<size_t>(_segments.size()) < _manifest.size()) {
+    static_assert(std::is_same_v<decltype(_segments.size()), size_t>);
+    static_assert(std::is_same_v<decltype(_manifest.size()), size_t>);
+    if (_segments.size() < _manifest.size()) {
         update_segments_incrementally();
     }
     auto ot_state = ss::make_lw_shared<storage::offset_translator_state>(
[nwatkins@fedora clang]$ ninja src/v/cloud_storage/CMakeFiles/v_cloud_storage.dir/remote_partition.cc.o 
[1/1] Building CXX object src/v/cloud_storage/CMakeFiles/v_cloud_storage.dir/remote_partition.cc.o
```

Reply:
something is definitely different about our environments because both are definitely size_t in my setup